http/relay/testdata/basic/app.js

```diff
@@ -1,51 +1,92 @@
-const http = require('http')
-const port = process.env.PORT
-
-http.createServer((req, res) => {
-  if (req.url.indexOf('/echo') == 0) {
-    const buffers = []
-    req.on('data', b => buffers.push(b))
-    req.on('end', _ => {
-      const body = Buffer.concat(buffers).toString()
-      res.setHeader('Content-Type', 'application/json')
-      res.end(JSON.stringify({
-        header: req.headers,
-        url: req.url,
-        body
-      }, null, 2))
-    })
-    return
+const http = require('http');
+const url = require('url');
+const qs = require('querystring');
+const port = process.env.PORT;
+
+let server;
+
+const routes = {};
```

Comment:
refactored this without adding deps, it was getting complex.

added:
* `/env` to fetch a key from process.env
* `/close` to stop accepting connections without killing the server
* `/pid` to fetch the pid
* `/swallowSignals` to install signal handlers